lib/registration.pm

```diff
@@ -174,25 +174,31 @@ sub add_suseconnect_product {
     $version //= '${VERSION_ID}';
     $arch //= '${CPU}';
     $params //= '';
-    $retry //= 0;    # run SUSEConnect a 2nd time to workaround the gpg error due to missing repo key on 1st run
+    $retry //= 3;    # run SUSEConnect $retry times before failing
 
     # some modules on sle12 use major version e.g. containers module
     my $major_version = '$(echo ${VERSION_ID}|cut -c1-2)';
     $version = $major_version if $name eq 'sle-module-containers' && is_sle('<15');
-
-    my $result = script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
-    if ($result != 0 && $retry) {
-        if ($name =~ /PackageHub/) {
-            record_soft_failure 'bsc#1124318 - Fail to get module repo metadata - running the command again as a workaround';
-
+    record_info('SCC product', "Activating product $name/$version/$arch $params");
+
+    my $try_cnt = 0;
+    my $result;
+    while ($try_cnt++ < $retry) {
+        eval {
+            assert_script_run("SUSEConnect -p $name/$version/$arch $params", timeout => $timeout);
+        };
+        if ($@) {
+            record_info('retry', "SUSEConnect failed to activate the module $name. Retrying...");
+            sleep 60;
         }
-
-        my $fail_message = ($name =~ /PackageHub/ && check_var('BETA', '1')) ? "PackageHub installation might fail in early development" : undef;
-        assert_script_run("SUSEConnect -p $name/$version/$arch $params", $timeout, $fail_message);
-
-    } elsif ($result && !$retry) {
-        die "SUSEConnect failed activating module $name with exit code (see log output): $result.";
+        else {
+            return 1;
```

Comment:
I´d add an info window here (you know, thinking about the reviewers) that there has been too many trials, so aborting.

Reply:
Isn't the line 201 enough?

Reply:
No, no, but this line is to return if the command succeded.. something like returning 0 in bash, but in perl we need to return "true" if ok. This condition will hit when `$@` is `undef`, which means that there are no errors from the `eval` section.

Reply:
@jlausuch exactly, I need more waay more coffee today xD